File: Preparation/week5/maxMIn.js
function maxMin(k, arr) {
  // Write your code here
  arr.sort((a, b) => a - b);
  let ret = Number.MAX_SAFE_INTEGER
  let i = 0;
  while(i <= arr.length - k){
      const min = arr[i]
      const max = arr[i + k - 1]
      if(ret > max - min){
          ret = max - min
      }
      i++
  }
  return ret
}
const k = 3;
const arr = [100, 200, 300, 350, 400, 401, 402];

maxMin(k, arr);

File: Preparation/week5/maxMIn.cpp
#include <algorithm>
#include <cmath>
#include <cstdio>
#include <iostream>
#include <string>
#include <unordered_map>
#include <vector>

using namespace std;
int maxMin(int k, vector<int> arr) {
  sort(arr.begin(), arr.end());
  int ret = INT_MAX;
  int i = 0;
  while (i <= arr.size() - k) {
    if (ret > arr[i + k - 1] - arr[i]) {
      ret = arr[i + k - 1] - arr[i];
    }
    i++;
  }
  return ret;
}